### ISSUE TITLE
no-constant-condition regression in v5.12.1

### ISSUE THREAD

**Issue by asapach** [eslint/eslint#11306]
**Tell us about your environment**

* **ESLint Version:** v5.12.1
* **Node Version:** v10.15.0
* **npm Version:** v6.4.1

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
    "rules": {
        "no-constant-condition": 2
    }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
if ((foo || "bar") === "baz") {
   // fail
}
```
[Demo](https://eslint.org/demo/#eyJ0ZXh0IjoiaWYgKChmb28gfHwgXCJiYXJcIikgPT09IFwiYmF6XCIpIHtcclxuICAgLy8gZmFpbFxyXG59Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo1LCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7Im5vLWNvbnN0YW50LWNvbmRpdGlvbiI6Mn0sImVudiI6e319fQ==)

```bash
eslint test.js
```

**What did you expect to happen?**
No lint errors, same as in v5.12.0

**What actually happened? Please include the actual, raw output from ESLint.**
`1:5 - Unexpected constant condition. (no-constant-condition)`

It seems to be caused by #11253.

**Are you willing to submit a pull request to fix this bug?**
maybe

**Comment by not-an-aardvark** [comment]
Thanks for the report, this looks like a bug to me.
